src/Ocelot/Request/Middleware/DownstreamRequest.cs

```diff
@@ -32,7 +32,9 @@ public DownstreamRequest(HttpRequestMessage request)
 
     public int Port { get; set; }
 
-    public string AbsolutePath { get; set; }
+    public string Path { get; set; }
+
+    public string ServicePathPrefix { get; set; } = string.Empty;
```

Comment:
To address the issue effectively, you have to share your solution by uploading it to GitHub. Introducing new property should be justified by a compelling reason, particularly when the existing implementation fails to resolve the specific use case.

Reply:
I cannot upload our custom service discovery implementation to github. Basically our service discovery returns an Uri, which is passed to ocelot as follows:

```csharp
private static List<Service> AsServiceList(ImmutableArray<ServiceInstance> serviceInstances)
{
    return serviceInstances.Select(item =>
    {
        var uri = new Uri(item.Uri, UriKind.Absolute);
        return new Service(item.ServiceName, 
            new ServiceHostAndPort(uri.Host, uri.Port, uri.Scheme, /* This is the new "prefix" */ uri.AbsolutePath), 
            item.Id, 
            "v1", 
            []);
    }).ToList();
}
```